src/pretix/control/forms/filter.py

```diff
@@ -811,6 +811,211 @@ def meta_properties(self):
+            ), get_current_timezone())
+            qs = qs.filter(created__lt=date_end)
+
+        if fdata.get('completed_from'):
+            date_start = make_aware(datetime.combine(
+                fdata.get('completed_from'),
+                time(hour=0, minute=0, second=0, microsecond=0)
+            ), get_current_timezone())
+            qs = qs.filter(payment_date__gte=date_start)
+
+        if fdata.get('completed_until'):
+            date_end = make_aware(datetime.combine(
+                fdata.get('completed_until') + timedelta(days=1),
+                time(hour=0, minute=0, second=0, microsecond=0)
+            ), get_current_timezone())
+            qs = qs.filter(payment_date__lt=date_end)
+
+        if fdata.get('event'):
+            qs = qs.filter(order__event=fdata.get('event'))
+
+        if fdata.get('organizer'):
+            qs = qs.filter(order__event__organizer=fdata.get('organizer'))
+
+        if fdata.get('state'):
+            qs = qs.filter(state=fdata.get('state'))
+
+        if fdata.get('provider'):
+            qs = qs.filter(provider=fdata.get('provider'))
+
+        if fdata.get('query'):
+            u = fdata.get('query')
+
+            matching_invoices = Invoice.objects.filter(
+                Q(invoice_no__iexact=u)
+                | Q(invoice_no__iexact=u.zfill(5))
+                | Q(full_invoice_no__iexact=u)
+            ).values_list('order_id', flat=True)
+
+            matching_invoice_addresses = InvoiceAddress.objects.filter(
+                Q(
+                    Q(name_cached__icontains=u) | Q(company__icontains=u)
+                )
+            ).values_list('order_id', flat=True)
+
+            if "-" in u:
+                code = (Q(order__event__slug__icontains=u.rsplit("-", 1)[0])
+                        & Q(order__code__icontains=Order.normalize_code(u.rsplit("-", 1)[1])))
+            else:
+                code = Q(order__code__icontains=Order.normalize_code(u))
+
+            matching_orders = OrderPayment.objects.filter(
+                Q(
+                    code
+                    | Q(order__email__icontains=u)
+                    | Q(order__comment__icontains=u)
+                )
+            ).values_list('id', flat=True)
+
+            mainq = (
+                Q(order__id__in=matching_invoices)
+                | Q(order__id__in=matching_invoice_addresses)
+                | Q(pk__in=matching_orders)
+            )
+
+            qs = qs.filter(mainq)
+
+        if fdata.get('amount'):
+            amount = fdata.get('amount')
+
+            def is_decimal(value):
+                result = True
+                parts = value.split('.', maxsplit=1)
+                for part in parts:
+                    result = result & part.isdecimal()
+                return result
+
+            if is_decimal(amount):
+                qs = qs.filter(amount=Decimal(amount))
+
+        if fdata.get('ordering'):
```

Comment:
We need an else branch here, currently the default order is pretty useless. I propose ``-created`` as the default order, as ``-payment_date`` will cause issues with null values.